include/qore/intern/QoreClassIntern.h

```diff
@@ -1028,86 +1029,53 @@ typedef std::map<char*, QoreVarInfo*, ltstr> var_map_t;
 template <typename T>
 class QoreMemberMapBase {
 public:
-    typedef std::pair<char*, T*> list_element_t;
-    typedef std::vector<list_element_t> member_list_t;
-    typedef typename member_list_t::const_iterator DeclOrderIterator;
-    // we use a vector map as the number of members is generally relatively small
-    typedef vector_map_t<char*, T*> member_map_t;
-    /*
-#ifdef HAVE_QORE_HASH_MAP
-    typedef HASH_MAP<char*, T*, qore_hash_str, eqstr> member_map_t;
-#else
-    typedef std::map<char*, T*, ltstr> member_map_t;
-#endif
-    */
-    typedef typename member_map_t::const_iterator SigOrderIterator;
+    typedef std::pair<char*, T*> member_list_element_t;
+    typedef std::vector<member_list_element_t> member_list_t;
+    typedef typename member_list_t::iterator iterator;
+    typedef typename member_list_t::const_iterator const_iterator;
+    member_list_t member_list;
 
-public:
     DLLLOCAL ~QoreMemberMapBase() {
-        for (typename member_map_t::iterator i = map.begin(), e = map.end(); i != e; ++i) {
+        for (auto& i : member_list) {
             //printd(5, "QoreMemberMap::~QoreMemberMap() this: %p freeing member %p '%s'\n", this, i->second, i->first);
-            delete i->second;
-            free(i->first);
+            delete i.second;
+            free(i.first);
```

Comment:
done except the `char*` issue, because this is allocated by the scanner, and changing this would require massive changes all over Qore - this needs to be changed, but in a major overhaul of memory management in the scanner